networkx/classes/multigraph.py

```diff
@@ -755,25 +755,33 @@ def degree_iter(self, nbunch=None, weight=None):
         [(0, 1), (1, 2)]
 
         """
+        if nbunch in self:
+            nbrs = self.adj[nbunch]
+            if weight is None:
+                return sum(len(self.adj[nbunch][node]) for node in self.adj[nbunch])
+            return (sum(sum(self.adj[nbunch][nbr][node].get(weight, 1) for node in self.adj[nbunch][nbr]) for nbr in self.adj[nbunch]))
```

Comment:
You compute nbrs but then don't use it. I think it would be easier to read (and maybe faster) if you use nbrs in place of self.adj[nbunch] in lines 761 and 762.